ScienceDirect.js

```diff
@@ -128,6 +128,10 @@ function parseIntermediatePDFPage(url, onDone) {
 			// Strip '0;URL='
 			var matches = pdfURL.match(/\d+;URL=(.+)/);
 			pdfURL = matches ? matches[1] : null;
+		} else {
```

Comment:
Could we check if the URL actually looks like a PDF URL? Will save us one attempted download

Reply:
What would you suggest to check? We can check whether `.pdf` is part of the url, also I am not sure that this is always the case. It should work here, because the pdfUrl is http://ac.els-cdn.com/S2095311916614284/1-s2.0-S2095311916614284-main.pdf?_tid=8fafcc14-9197-11e7-ac86-00000aab0f02&acdnat=1504546759_dcbe1d593fe692f3c36d4817dad94803

You are right, that possibly we try to download something which is not the correct url for the pdf. On the other side, we would have a possible (else) case where nothing happens. I am not sure, what is better/worse.

Reply:
Yes, I was going to look for ".pdf?" -- I've looked at a large number of these, I think that's always in there.

Reply:
Okay, will add such a test (but maybe exclude the question mark?).